Infinite Eagle Eye & Super Damage

Deadlineem opened the pull request:
This PR adds a toggleable option for Eagle Eye.

This checks to see if eagle eye is active, if its not, it activates it for you then sets eagle eye to run for an infinite duration.

NOTE:  While eagle eye is active, when you try to sprint, it deactivates, this will keep reactivating it for you but will pretty much nullify sprinting.

If this is an issue, removing `PLAYER::_SECONDARY_SPECIAL_ABILITY_SET_ACTIVE(PLAYER::PLAYER_ID());` will only set the infinite eagle eye duration, but it must be manually activated, **_sprinting is still going to disable eagle eye._**

Disabling will remove all effects (Eagle eye, infinite duration) one time only.

##Update 2:

Adds a togglable option for super damage

Closes https://github.com/YimMenu/HorseMenu/issues/115 -- Infinite Eagle Eye

Closes https://github.com/YimMenu/HorseMenu/issues/104 -- Instant Kill

Closes https://github.com/YimMenu/HorseMenu/issues/132 -- Fishing bug

Rxann commented on src/game/features/self/EagleEye.cpp:
```diff
@@ -0,0 +1,33 @@
+
+		virtual void OnTick() override
+		{
+			if(!PLAYER::_IS_SECONDARY_SPECIAL_ABILITY_ACTIVE(PLAYER::PLAYER_ID())) // Checks if Eagle Eye is active before toggling it on
```
I think this could be modified to detect when they are in eagle eye and just give them unlimited then

Deadlineem replied:
> I think this could be modified to detect when they are in eagle eye and just give them unlimited then

removing the if statement and its innards does that on its own, but it wont auto activate for you.  But yeah it works that way by removing this

```if(!PLAYER::_IS_SECONDARY_SPECIAL_ABILITY_ACTIVE(PLAYER::PLAYER_ID())) // Checks if Eagle Eye is active before toggling it on
            {
                PLAYER::_SECONDARY_SPECIAL_ABILITY_SET_ACTIVE(PLAYER::PLAYER_ID()); // Toggles Eagle Eye on
                Notifications::Show("Eagle Eye", "Infinite Eagle eye has been enabled.", NotificationType::Success);
                Notifications::Show("WARNING", "When this is active, it will always reactivate eagle eye for you, you still cannot run using it.", NotificationType::Warning);
            }```

Deadlineem replied:
Only using toggle for `PLAYER::_MODIFY_INFINITE_TRAIL_VISION(PLAYER::PLAYER_ID(), 1); // Sets Eagle Eye to Infinite` will toggle infinite eagle eye on/off, youd still have to activate it on your own.

Deadlineem commented:
@DayibBaba whats your take, should I remove the check if eagle eye is active and just set the infinite modifier, let them activate eagle eye on their own, or is this ok?

Rxann commented:
> @DayibBaba whats your take, should I remove the check if eagle eye is active and just set the infinite modifier, let them activate eagle eye on their own, or is this ok?

I am not Dayibaba but i think you should let them activate it on your own, but by all means it is up to you and dayibaba. just a suggestion.

DayibBaba commented:
We should investigate why sprinting and deadeye are mutualy exclusive. I don't suppose logically they would, a simple nop could make this a much more elegant feature.

Deadlineem commented:
> We should investigate why sprinting and deadeye are mutualy exclusive. I don't suppose logically they would, a simple nop could make this a much more elegant feature.

I agree with you on that, it bothers me that you cant run with eagle eye active.  Same with deadeye, but while i was working on the eagle eye feature i found one for deadeye as well, so really, its mainly about the running now.

Deadlineem commented:
@DayibBaba I found something related to motion states in natives which brought me here

https://github.com/Halen84/RDR3-Native-Flags-And-Enums/tree/main/CPedMotionStates__eMotionState

This might help?

Rxann commented:
> We should investigate why sprinting and deadeye are mutualy exclusive. I don't suppose logically they would, a simple nop could make this a much more elegant feature.

Would we be patching an instruction in a native function?

Rxann commented on src/game/features/players/toxic/LightningStrike.cpp:
```diff
@@ -0,0 +1,22 @@
+		virtual void OnCall(Player player) override
+		{
+			auto playerCoords = player.GetPed().GetPosition();
+			FIRE::ADD_EXPLOSION(playerCoords.x, playerCoords.y, playerCoords.z, 33, 5.0f, true, false, 5.0f);
```
I think there is an explosion type enum that can be used here

Rxann replied:
ExplosionTypes:: SPARKS_WITH_FIRE_AFTERWARDS might be what is best, i have not tested though

Deadlineem replied:
Altered, thanks!  i wasnt sure which one to use lol

Deadlineem replied:
> ExplosionTypes:: SPARKS_WITH_FIRE_AFTERWARDS might be what is best, i have not tested though

That explosion type is not the type for lightning sparks, i did test that and it was not the same explosion type so this was reverted @Rxann

DayibBaba commented on src/game/features/self/NoSpread.cpp:
```diff
@@ -0,0 +1,20 @@
+
+namespace YimMenu::Features
+{
+	//WEAPON::_SET_PED_INFINITE_AMMO_CLIP Changes too many things to be considered, such as granting all ammo types.
```
This looks like an artifact from another command.

Deadlineem replied:
Infinite clip, its what nospread was based from but that commented line is unneeded

DayibBaba commented on src/game/features/self/SuperDamage.cpp:
```diff
@@ -0,0 +1,30 @@
+
+		virtual void OnTick() override
+		{
+				const float DAMAGE_MULT = 100000000.0;
```
If there is care for however high this is, you can use `std::numeric_limits<float>::max();`

Deadlineem replied:
eventually im would like to rework this as an input value or slider

DayibBaba commented on src/util/Rewards.cpp:
```diff
@@ -102,28 +102,37 @@ namespace YimMenu::Rewards
+	
 				Vector3 coords = ENTITY::GET_ENTITY_COORDS(PLAYER::PLAYER_PED_ID(), true, false);
-				Object obj = OBJECT::CREATE_OBJECT(hash, coords.x, coords.y, coords.z, true, NETWORK::NETWORK_IS_HOST_OF_THIS_SCRIPT(), 1, 0, 1);
+				float forwardX = ENTITY::GET_ENTITY_FORWARD_X(PLAYER::GET_PLAYER_PED_SCRIPT_INDEX(PLAYER::PLAYER_ID()));
```
You can use `GET_OFFSET_FROM_ENTITY_IN_WORLD_COORDS` and set the Y coordinate to be 2 paces in front.

Deadlineem replied:
Either/or will work.  Ive never used that native before so i did not know about this one lol

Deadlineem replied:
Fixed this, ready for review of the rest and merge!

DayibBaba commented:
> > We should investigate why sprinting and deadeye are mutualy exclusive. I don't suppose logically they would, a simple nop could make this a much more elegant feature.
> 
> Would we be patching an instruction in a native function?

Im affraid it runs a little deeper than that.

Deadlineem commented:
All of the UI changes can be seen here and some of the new options can also be seen

Title bar is dropdown capable, this is more for when you want to see something like how your ped model looks when changed, instead of closing the menu, you can just pull it closed with the arrow.  This also seems to help the columns display better.

Build version uses the DEBUG:: native to get the game build information.

![Screenshot_32](https://github.com/YimMenu/HorseMenu/assets/24372625/e1395e75-7feb-4c4a-9286-bd774e2aaeca)

Deadlineem commented on README.md:
```diff
@@ -2,14 +2,19 @@
-**Current Look (Outdated)**
+## Screenshots / UI Design
+
+![image](https://cdn.discordapp.com/attachments/1095152109445451808/1224394225689301073/Screenshot_32.png?ex=661d54fb&is=660adffb&hm=281fa23762574f9b03c1cdffbd7c509f8202c7b9dfc9a45f12026b563f1a71b3&)
```
the image posted in the comments should be a github link, this can be changed before pushing the PR.

Rxann commented on src/game/features/self/EagleEye.cpp:
```diff
@@ -0,0 +1,33 @@
+
+		virtual void OnTick() override
+		{
+			if(!PLAYER::_IS_SECONDARY_SPECIAL_ABILITY_ACTIVE(PLAYER::PLAYER_ID())) // Checks if Eagle Eye is active before toggling it on
```
YimMenu::Self::Id could be used instead of PLAYER:PLAYER_ID

Rxann commented on src/game/features/self/EagleEye.cpp:
```diff
@@ -0,0 +1,33 @@
+		{
+			if(!PLAYER::_IS_SECONDARY_SPECIAL_ABILITY_ACTIVE(PLAYER::PLAYER_ID())) // Checks if Eagle Eye is active before toggling it on
+			{
+				PLAYER::_SECONDARY_SPECIAL_ABILITY_SET_ACTIVE(PLAYER::PLAYER_ID()); // Toggles Eagle Eye on
```
Here aswell

Rxann commented on src/game/features/self/EagleEye.cpp:
```diff
@@ -0,0 +1,33 @@
+				Notifications::Show("Eagle Eye", "Infinite Eagle eye has been enabled.", NotificationType::Success);
+				Notifications::Show("WARNING", "When this is active, it will always reactivate eagle eye for you, you still cannot run using it.", NotificationType::Warning);
+			}
+			PLAYER::_MODIFY_INFINITE_TRAIL_VISION(PLAYER::PLAYER_ID(), 1); // Sets Eagle Eye to Infinite
```
Here

Rxann commented on src/game/features/self/EagleEye.cpp:
```diff
@@ -0,0 +1,33 @@
+
+		virtual void OnDisable() override
+		{
+			PLAYER::_SECONDARY_SPECIAL_ABILITY_SET_DISABLED(PLAYER::PLAYER_ID(), 1); // Disables Eagle Eye
```
And here and the line below.

Rxann reviewed (commented):
I could have put this in one comment but...

Deadlineem reviewed (commented):
changes made

Rxann commented:
> changes made

Alright cool

Deadlineem reviewed (commented):
Changed the screenshot link to a github directed link

>> https://github.com/YimMenu/HorseMenu/assets/24372625/e1395e75-7feb-4c4a-9286-bd774e2aaeca

Rxann reviewed (commented):
Code looks good to me, but we should still look into what was mentioned above about patching the sprint cancelling eagle eye.

DayibBaba commented:
Much like `Self::PlayerPed`  we also have `Self::Veh` and `Self::Mount`. Please use these wherever possible.

Deadlineem commented:
> Much like `Self::PlayerPed` we also have `Self::Veh` and `Self::Mount`. Please use these wherever possible.

Ill make all these changes to the scripts today and push the updates.  This merge is pretty big lol

Deadlineem commented:
Reverted HorseGodmode.cpp to its original state.  This may also no longer be needed in general if you push PR #130 with his VehicleGodmode.cpp, that should logically get the vehicle the ped is using (whether its a mount or a wagon) and apply godmode to them.

Deadlineem reviewed (commented):
Changed a few things to resolve issues raised in the scripts, this should be ready for merge now.

DayibBaba commented:
Will be testing the file before reviewing again.

Deadlineem commented:
> Will be testing the file before reviewing again.

Let me know if there is anything else, thanks for your time!

Deadlineem commented:
> > > Will be testing the file before reviewing again.
> > 
> > 
> > Let me know if there is anything else, thanks for your time!
> 
> Super Damage does not seem to work.

Mostly it does damage with melee (like being able to kick an alligator and it dies) on peds and players, it does 3/4 of a health bar of damage.  The weapons seem to have the same results.  It did work well before adding `std::numeric_limits<float>::max()`

DayibBaba commented:
> > > > Will be testing the file before reviewing again.
> > > 
> > > 
> > > Let me know if there is anything else, thanks for your time!
> > 
> > 
> > Super Damage does not seem to work.
> 
> Mostly it does damage with melee (like being able to kick an alligator and it dies) on peds and players, it does 3/4 of a health bar of damage. The weapons seem to have the same results. It did work well before adding `std::numeric_limits<float>::max()`
> 
> However, this still could be a user input box defaulted to 1.0 like yimmenu has

The high values might cause some sort of treshold, try lower values instead of the max.

Deadlineem commented:
> > > > > Will be testing the file before reviewing again.
> > > > 
> > > > 
> > > > Let me know if there is anything else, thanks for your time!
> > > 
> > > 
> > > Super Damage does not seem to work.
> > 
> > 
> > Mostly it does damage with melee (like being able to kick an alligator and it dies) on peds and players, it does 3/4 of a health bar of damage. The weapons seem to have the same results. It did work well before adding `std::numeric_limits<float>::max()`
> > However, this still could be a user input box defaulted to 1.0 like yimmenu has
> 
> The high values might cause some sort of treshold, try lower values instead of the max.

I set it to 1 mill on each, removed Melee so it can be separated from this and added another native `NETWORK::_SET_LOCAL_PLAYER_DAMAGE_MULTIPLIER_FOR_PLAYER`, i do not have time to test new native till after work tonight, i am installing RDR2 again rn, had a PC issue.  If you wouldn't mind giving it a test and letting me know, id appreciate it @Rxann or @DayibBaba

Deadlineem commented:
@DayibBaba added a check onto Infinite Clip to check if the held weapon is a fishing rod, if it is, it doesn't add infinite clip to it as it was breaking the fishing system and not allowing users to cast the rod out, see #132

Deadlineem commented:
Closes #132

Deadlineem commented:
@DayibBaba I wont push another PR until this one gets merged.  Still trying to learn how to push different PR's, think im gonna try github desktop.

MarkEcza commented:
@Deadlineem status?

DayibBaba commented:
> Should be done and ready to merge, I'm not working on more PR's till this ones merged bc i cant seem to figure out how to make separate PR's lmao

You need to use different branches to make multiple PRs.

Deadlineem commented:
> > Should be done and ready to merge, I'm not working on more PR's till this ones merged bc i cant seem to figure out how to make separate PR's lmao
> 
> You need to use different branches to make multiple PRs.

Ill talk with rxan on discord and see if he can give me a visual on how to do it properly after this is merged.  I would also like it if someone could take a stab at allowing the user to sprint while this is toggled, but for now i think we can push this until we figure it out.

Also if you guys have discord, feel free to add me there to collaborate this project and even yimmenu itself, i do use that as well.

Rxann commented on src/game/features/self/SuperDamage.cpp:
```diff
@@ -0,0 +1,29 @@
+			// Set all player weapon damage modifiers to their defaults
+			PLAYER::SET_PLAYER_WEAPON_DAMAGE_MODIFIER(Self::PlayerPed, 1.0);
+			PLAYER::_SET_PLAYER_EXPLOSIVE_WEAPON_DAMAGE_MODIFIER(Self::PlayerPed, 1.0);
+			NETWORK::_SET_LOCAL_PLAYER_DAMAGE_MULTIPLIER_FOR_PLAYER(Self::PlayerPed, 1.0)
```
`;` needed at the end of the native call.

Rxann commented on src/game/features/self/SuperDamage.cpp:
```diff
@@ -0,0 +1,29 @@
+		{
+			PLAYER::SET_PLAYER_WEAPON_DAMAGE_MODIFIER(Self::PlayerPed, 1000000.0);
+			PLAYER::_SET_PLAYER_EXPLOSIVE_WEAPON_DAMAGE_MODIFIER(Self::PlayerPed, 1000000.0);
+			NETWORK::_SET_LOCAL_PLAYER_DAMAGE_MULTIPLIER_FOR_PLAYER(Self::PlayerPed, 1000000.0)
```
`;` needed at the end of the native call

Rxann commented on src/game/frontend/submenus/Network.cpp:
```diff
@@ -26,7 +26,8 @@ namespace YimMenu::Submenus
 		auto spoofingColumns  = std::make_shared<Column>(1);
-		session->AddItem(std::make_shared<CommandItem>("explodeall"_J));
+		session->AddItem(std::make_shared<CommandItem>("quitsession"_J));
+    session->AddItem(std::make_shared<CommandItem>("explodeall"_J));
```
This might have been when I was resolving conflicts, but if this could be indented that would be great.

Deadlineem commented:
@Rxann I commented out the eagle eye option in Self.cpp for now, this way we can leave the feature in there until we figure out the run situation and then just uncomment it later, that should suffice for now to be able to merge the PR as it is.

Rxann commented on src/core/frontend/manager/UIManager.cpp:
```diff
@@ -31,7 +32,12 @@ namespace YimMenu
 		ImGui::EndChild();
 		
-		ImGui::Text("HorseMenu");
+		std::string gameVer = DEBUG::GET_GAME_VERSION_NAME();
```
Are you sure this works? Calling natives in the rendering thread doesn't usually work.

Deadlineem replied:
1000% sure it works as expected unless the game changed since adding this.  It shows the game version at the bottom left of the gui.

FIying-Scotsman replied:
> Are you sure this works? Calling natives in the rendering thread doesn't usually work.

Natives which just return class members (i.e GET_PLAYER_NAME just returns CName) can be called on the DX thread as those don't require TLS. It's not good practice, but can be done. Other natives which need a script thread TLS will still crash when called on the DX thread.

I think GET_GAME_VERSION just prints whatever value is in the update.rpf gameversion.txt if I remember right so won't need a script's TLS given it's a file read operation.

Deadlineem commented:
> What's the hold up? This has been open for 3 months...

The infinite eagle eye toggles itself off when you run, so it auto toggles back on but disables running movements while its active.  Dayibaba said it could be made into a more elegant feature if we figure out how to bypass the fact that when eagle eye is on, the running cancels it, thats the only thing this PR needs to push and rxan and myself have no clue how to fix that factor

Rxann commented:
> > > What's the hold up? This has been open for 3 months...
> > 
> > 
> > The infinite eagle eye toggles itself off when you run, so it auto toggles back on but disables running movements while its active.  Dayibaba said it could be made into a more elegant feature if we figure out how to bypass the fact that when eagle eye is on, the running cancels it, thats the only thing this PR needs to push and rxan and myself have no clue how to fix that factor
> 
> That's because eagle eye plus is not activated. Try the native _EAGLE_EYE_SET_PLUS_FLAG_DISABLED

Yes, try this.

Deadlineem commented:
> > > What's the hold up? This has been open for 3 months...
> > 
> > 
> > The infinite eagle eye toggles itself off when you run, so it auto toggles back on but disables running movements while its active.  Dayibaba said it could be made into a more elegant feature if we figure out how to bypass the fact that when eagle eye is on, the running cancels it, thats the only thing this PR needs to push and rxan and myself have no clue how to fix that factor
> 
> That's because eagle eye plus is not activated. Try the native _EAGLE_EYE_SET_PLUS_FLAG_DISABLED

Ill give this a try in a few minutes and test if anything changes, if so ill push to the PR to make the changes so this can be merged, I honestly miss these features lmao

Deadlineem commented:
@Rxann This is ready for merge now, Eagle eye now allows you to run while its active, you can also use it on your horse.

Rxann commented:
I will test this soon

Deadlineem commented:
> > I will test this soon
> 
> _cough cough_

I know right?!  @Rxann tick tock im not getting any younger!  I can push the train speed modifications next after this PR and work on the ptfx as well

Rxann commented:
> > > I will test this soon
> 
> > 
> 
> > _cough cough_
> 
> 
> 
> I know right?!  @Rxann tick tock im not getting any younger!  I can push the train speed modifications next after this PR and work on the ptfx as well

My fault, been busy on other horsemenu stuff

Deadlineem commented:
> > > > I will test this soon
> > 
> > 
> > > 
> > 
> > 
> > > _cough cough_
> > 
> > 
> > I know right?!  @Rxann tick tock im not getting any younger!  I can push the train speed modifications next after this PR and work on the ptfx as well
> 
> My fault, been busy on other horsemenu stuff

I wanna be busy with it too!  Merge teh PR brotha!

Rxann commented:
> > > > > I will test this soon
> 
> > > 
> 
> > > 
> 
> > > > 
> 
> > > 
> 
> > > 
> 
> > > > _cough cough_
> 
> > > 
> 
> > > 
> 
> > > I know right?!  @Rxann tick tock im not getting any younger!  I can push the train speed modifications next after this PR and work on the ptfx as well
> 
> > 
> 
> > My fault, been busy on other horsemenu stuff
> 
> 
> 
> I wanna be busy with it too!  Merge teh PR brotha!

Have u tested it?

Deadlineem commented:
> > > > > > I will test this soon
> > 
> > 
> > > >
> > 
> > 
> > > >
> > 
> > 
> > > > >
> > 
> > 
> > > >
> > 
> > 
> > > >
> > 
> > 
> > > > > _cough cough_
> > 
> > 
> > > >
> > 
> > 
> > > >
> > 
> > 
> > > > I know right?!  @Rxann tick tock im not getting any younger!  I can push the train speed modifications next after this PR and work on the ptfx as well
> > 
> > 
> > > 
> > 
> > 
> > > My fault, been busy on other horsemenu stuff
> > 
> > 
> > I wanna be busy with it too!  Merge teh PR brotha!
> 
> Have u tested it?

Yes, it works as it should, i haven't noticed any bugs with it yet, the rest of them were tested awhile ago.

Rxann commented:
> > > > > > > I will test this soon
> 
> > > 
> 
> > > 
> 
> > > > >
> 
> > > 
> 
> > > 
> 
> > > > >
> 
> > > 
> 
> > > 
> 
> > > > > >
> 
> > > 
> 
> > > 
> 
> > > > >
> 
> > > 
> 
> > > 
> 
> > > > >
> 
> > > 
> 
> > > 
> 
> > > > > > _cough cough_
> 
> > > 
> 
> > > 
> 
> > > > >
> 
> > > 
> 
> > > 
> 
> > > > >
> 
> > > 
> 
> > > 
> 
> > > > > I know right?!  @Rxann tick tock im not getting any younger!  I can push the train speed modifications next after this PR and work on the ptfx as well
> 
> > > 
> 
> > > 
> 
> > > > 
> 
> > > 
> 
> > > 
> 
> > > > My fault, been busy on other horsemenu stuff
> 
> > > 
> 
> > > 
> 
> > > I wanna be busy with it too!  Merge teh PR brotha!
> 
> > 
> 
> > Have u tested it?
> 
> 
> 
> Yes, it works as it should, i haven't noticed any bugs with it yet, the rest of them were tested awhile ago.

Once you fix what MBG reviewed and he confirms it's fine, it can be merged.

Deadlineem commented on src/game/features/self/EagleEye.cpp:
```diff
@@ -0,0 +1,34 @@
+			{
+				PLAYER::_SECONDARY_SPECIAL_ABILITY_SET_ACTIVE(YimMenu::Self::Id); // Toggles Eagle Eye on
+				PLAYER::_EAGLE_EYE_SET_PLUS_FLAG_DISABLED(YimMenu::Self::Id, false); // Allows running while eagle eye is active
+				Notifications::Show("Eagle Eye", "Infinite Eagle eye has been enabled.  This allows you to run while eagle eye is active as well.", NotificationType::Success);
```
Nope, it was originally there because of the run portion not working, i forgot to remove it.

Deadlineem commented on src/game/features/network/QuitSession.cpp:
```diff
@@ -0,0 +1,20 @@
+	{
+		using Command::Command;
+
+		virtual void OnCall() override
```
Can be removed if its already added, pr is 3 months old, this would have been here first lol

Rxann replied:
We have `SCRIPTS::BAIL_TO_LANDING_PAGE` and not `NETWORK::NETWORK_SESSION_LEAVE_SESSION()`. I don't know if there is a huge difference, if any at all but it could be considered.

Deadlineem commented on src/game/features/self/EagleEye.cpp:
```diff
@@ -0,0 +1,34 @@
+			PLAYER::_SECONDARY_SPECIAL_ABILITY_SET_DISABLED(YimMenu::Self::Id, 1); // Disables Eagle Eye
+			PLAYER::_MODIFY_INFINITE_TRAIL_VISION(YimMenu::Self::Id, 0);           // Turns off Infinite Eagle Eye
+			PLAYER::_EAGLE_EYE_SET_PLUS_FLAG_DISABLED(YimMenu::Self::Id, NULL); //Setting to truee breaks it, leave it at NULL
+			Notifications::Show("Eagle Eye", "Infinite Eagle eye has been disabled.", NotificationType::Success);
```
Same as before, prolly not.  Debug things

Deadlineem commented:
> > I wanna be busy with it too! Merge teh PR brotha!
> 
> You can create a different branch while waiting for your existing PR to get merged. You can even open multiple PRs at the same time

I tried figuring it out, failed 100 times 😂

Deadlineem commented:
> > I tried figuring it out, failed 100 times 😂
> 
> https://github.com/Deadlineem/HorseMenu/branches [Create a new one](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-and-deleting-branches-within-your-repository) based on upstream (/YimMenu/HorseMenu) master

For 3 months ive been trying to figure it out, every time i would just use vs and commit i tried making a new branch in vs but not on github, im dumb 🤣

Rxann commented on src/game/frontend/Menu.cpp:
```diff
@@ -37,8 +37,10 @@ namespace YimMenu
-			    if (ImGui::Begin("HorseMenu", nullptr, ImGuiWindowFlags_NoDecoration))
+			    ImGui::SetNextWindowSize(ImVec2(780, 620 /*add auto resize*/), ImGuiCond_Once);
+				// Window default positioning, adjust as needed.
+				ImGui::SetNextWindowPos(ImVec2(0.5f, 0.5f), ImGuiCond_Once);
```
Indent this

Deadlineem commented:
@Rxann all issues resolved and tested/working

Rxann commented on src/game/features/self/NoSpread.cpp:
```diff
@@ -0,0 +1,24 @@
+		}
+	};
+
+	static AccuracyMax _AccuracyMax{"maxaccuracy", "No Spread", "Always perfect accuracy on shots"};
```
Lets rename this class to NoSpread instead of AccuracyMax

Deadlineem commented:
> LGTM
> 
> Also @Deadlineem did you ever get branches to work?

I haven't set any new ones up yet but after this PR i do have plans to add some train settings in. 

I did notice something a bit messed up though when i was working on this, the UI is a bit funky on certain tabs like the tab the player list is on, the box on the left with the IP and other info is too small in width, this is also the same in the recovery tab under self as well but not nearly as bad.

Also, i was playing around with the teleport to waypoint option if you set a waypoint too far it will continuously fail to load/teleport, commented out the if statement to load ground and commented the brackets around it, added the function 

LoadGroundAtCoords(coords); 

and it loads the teleport as it should no matter the distance but you spawn 1000 feet in the sky.  @Rxann @maybegreat48
